core/templates/dev/head/base.js

```diff
@@ -211,6 +213,17 @@ oppia.controller('Base', [
       return false;
     };
 
+    //  Method to check if the window size is narrow
+    $scope.isWindowNarrow = function() {
```

Comment:
Yeah. This is exactly the point what I was raising in my previous.
Yes, Two-way data binding will solve the problem of the number of calls.